contentcuration/contentcuration/views/internal.py

```diff
@@ -652,7 +652,14 @@ def create_node(node_data, parent_node, sort_order):
         except ObjectDoesNotExist:
             raise ObjectDoesNotExist("Invalid license found")
 
-    return ContentNode.objects.create(
+    tags = []
+    if 'tags' in node_data:
+        tag_data = node_data['tags']
+        if tag_data is not None:
+            for tag in tag_data:
+                tags.append(ContentTag.objects.get_or_create(tag_name=tag)[0])
```

Comment:
Note that ContentTag objects also accept the optional channel property, for tags "scoped" to a particular channel. Not sure if we want to set the channel attr or leave it NULL.

Reply:
Do you know what that scoping feature is used for? One issue I'm a bit concerned about is what happens when a second channel adds a tag that is already scoped to another channel. Does it create a duplicate, overwrite the channel property, or forbid it?

Reply:
It will create a new tag scoped to a different channel. I can't remember the full reason, but I think this was to prevent typos from making it into the global pool of tags (and allows for faster querying)

Reply:
Think it's more for the autocomplete and publishing functions- instead of getting a list of the full set of tags, it will only take those related to the channel and pass them to the page/save them to the generated Kolibri db

Reply:
Oops, I was mentally mixing 'Channel' with 'ContentNode' in my head, which is why I was wondering about why we both had ContentNode.tags and this ForeignKey relationship as well... 😓 Sorry about that! I'll go ahead and make the change and send an updated PR.